### ISSUE TITLE
Difference in Scss since 0.28.21 release

### ISSUE THREAD

**Issue by jerone** [Glavin001/atom-beautify#752]
Before 0.28.21 was released my beautified Scss (Sass) looked like this:
```scss
/* Imports */
@import "imports/_variables";
@import "imports/_mixins";
@import "imports/reset";
@import "imports/_button";

/* Source */
.content {
	background-color: $dark_brown;
}

form {
	height: 100%;

	.content {
		height: 100%;
	}
}
```

Now it looks like this (double new lines, extra line between `@imports`):
```scss
/* Imports */
@import "imports/_variables";
@import "imports/_mixins";
@import "imports/reset";

@import "imports/_button";
/* Source */
.content {
	background-color: $dark_brown;
}


form {
	height: 100%;


	.content {
		height: 100%;
	}

}
```

This will create unneeded differences in my commits.

***

_Atom 1.4.0_
_Atom Beautify 0.28.21_

**Comment by prettydiff** [comment]
I just updated to version v0.28.21 this morning and I am having trouble reproducing this defect using either Pretty Diff or CssComb as the beautifier.  The only defect I can reproduce is from Pretty Diff with the strangeness of the empty line appearing before the final import when it should be appearing after (before the comment).

**Comment by jerone** [comment]
I can confirm this now on two different machines (Windows 7 and Windows 8.1) with Atom 1.4.0 and 1.5.0-beta0. No `.jsbeautifyrc` file. Following Atom settings:
```cson
  "atom-beautify":
    _analyticsUserId: random
    css_convert_quotes: "double"
    css_newline_between_rules: true
    css_preserve_newlines: true
    css_selector_separator_newline: true
    js_end_with_newline: true
```

**Comment by prettydiff** [comment]
I am still having trouble reproducing the problem.  I think I have a solution, but I just need you to verify it.  Could you open the concerned file:  `~/.atom/packages/atom-beautify/node_modules/prettydiff/lib/csspretty.js`

And then replace https://github.com/prettydiff/prettydiff/blob/master/lib/csspretty.js#L1129-L1140 with:

    token[a] = lf + token[a];

If this works for you I can publish the update tonight.

**Comment by jerone** [comment]
Better, but still two issues:

* Two new lines between rules.
* Indentation issue before `.content {`.

**Result:**

```scss
/* Imports */
@import "imports/_variables";
@import "imports/_mixins";
@import "imports/reset";
@import "imports/_button";

/* Source */
.content {
	background-color: $dark_brown;
}


form {
	height: 100%;
	
	
.content {
		height: 100%;
	}
}
```

**Comment by prettydiff** [comment]
I will keep looking.

**Comment by prettydiff** [comment]
Got it.  It was interference with the `css_newline_between_rules: true` option.

If the entire block mentioned earlier is replaced with this:

                    if (lines[a] > 1 && (types[a] === "start" || types[a] === "end" || (types[a] === "selector" && options.cssinsertlines === false) || types[a] === "comment")) {
                        token[a] = lf + token[a];
                    }

An empty line should not be inserted before a selector if the mentioned option is enabled since its going to get an empty line anyways.

**Comment by prettydiff** [comment]
Pretty Diff resolution published as v1.16.12.
